returnn/compile.py

```diff
@@ -92,12 +94,13 @@ def run(self):
             "--train=%d" % self.train,
             "--eval=%d" % self.eval,
             "--search=%d" % self.search,
-            "--device=%s" % self.device,
             "--verbosity=%d" % self.verbosity,
             "--output_file=%s" % self.out_graph.get_path(),
             "--output_file_model_params_list=model_params",
             "--output_file_state_vars_list=state_vars",
         ]
+        if self.device is not None:
+          args.extend(["--device=%s" % self.device])
```

Comment:
Wrong amount of whitespace. 
Also why extend with a single item list when there is append directly below?